### PR TITLE
Data scouting classes and producers (76X)

### PR DESCRIPTION
**Pull request by duanders**
-Adding a package DataFormats/DataScouting with new scouting object classes (ScoutingCaloJet, ScoutingPFJet, ScoutingParticle, ScoutingVertex)

-Adding a package HLTrigger/DataScouting with producers for the scouting objects (ScoutingPFProducer, ScoutingCaloProducer)

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
Sorry, this is not acceptable for HLTrigger/ : two new packages for a few classes.
HLTrigger is object centric (electron, muon, jet).
Please put the classes into the existing HLTrigger/* package structure.
The producers must become multi-threading friendly.

**Comment by Martin-Grunewald** [comment]
Also, HLT classes (under HLTrigger) start their class name with HLT.

**Comment by duanders** [comment]
Hi @Martin-Grunewald 
I moved the producer classes into HLTrigger/JetMET and prepended HLT to their names.  They are now edm::stream::EDProducers.

Can the new products stay in DataFormats/DataScouting or is there another preferred place for them?

**Comment by Martin-Grunewald** [comment]
DataFormats/HLTReco

File: HLTrigger/JetMET/plugins/HLTScoutingCaloProducer.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+//         Created:  Fri, 12 Jun 2015 15:49:20 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
```
Comment (Martin-Grunewald) [inline HLTrigger/JetMET/plugins/HLTScoutingCaloProducer.cc]: I think this must be:

#include "FWCore/Framework/interface/stream/EDProducer.h"

**Comment by duanders** [comment]
@Martin-Grunewald 
I moved the new products into DataFormats/HLTReco as requested.  Also included stream/EDProducer.h in the producer classes.

**Comment by Martin-Grunewald** [comment]
OK, thanks. Let's see what jenkins tests have to say.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
Actually, how will these dataformat classes be used?
As they are new types, they will not be packed up in any trigger summary.
Also, the ScoutingParticle is identical to the TriggerObject already defined
in DataFormats/HLTReco. Finally, you did not prepend HLT to the dataformat
classes.
Is there a talk or presentation somewhere?

**Comment by Martin-Grunewald** [comment]
76X IBs are still broken, too bad!

**Comment by duanders** [comment]
Hi @Martin-Grunewald ,
These are for use with data scouting: https://indico.cern.ch/event/390107/session/7/contribution/39/attachments/780401/1070037/11052015scoutingPPDFollowUp.pdf.  

The purpose is to provide a set of bare-bones classes for packing the relevant data members from the HLT jet and PF candidate collections.  The producers for these scouting objects will run at the ends of the scouting HLT paths, and their output will be saved and stored in scouting datasets offline.  

It's important that the scouting classes do not change, except in totally backwards compatible ways, so that the scouting datasets can still be read long in the future.  Thus we want ScoutingParticle (or HLTScoutingParticle) to be a separate class from TriggerObject, in case that class might change someday.

**Comment by Martin-Grunewald** [comment]
@duanders 
This PR is waiting for your changes!

**Comment by Martin-Grunewald** [comment]
BTW, this must be backported to 75X and 74X for use i data taking.
I hope that the classversion numbers/hashes are unchanged?

**Comment by Martin-Grunewald** [comment]
please test

**Comment by duanders** [comment]
Hi @Martin-Grunewald ,
I changed the checksums and reset the class versions to 10 because the class names changed.  Is this the right procedure or should I have incremented them to version 11?  

I'll create the PRs for 75 and 74 if there are no further changes needed here for now.

**Comment by Dr15Jones** [comment]
If the class names are all new, then you should set the class version to 2. We only use 10 when we are dealing with a class we had previously stored without any version number.

**Comment by Martin-Grunewald** [comment]
please test
(the most recent IB should have the [unrelated] fix)

**Comment by Martin-Grunewald** [comment]
please test

File: HLTrigger/JetMET/plugins/HLTScoutingCaloProducer.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+HLTScoutingCaloProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("caloJetCollection",edm::InputTag("hltAK4CaloJets"));
+    desc.add<edm::InputTag>("vertexCollection", edm::InputTag("hltPixelVertices"));
+    desc.add<edm::InputTag>("metCollection", edm::InputTag("hltMetCleanUsingJetID"));
+    desc.add<edm::InputTag>("rho", edm::InputTag("hltFixedGridRhoFastjetAllCalo"));
+    desc.add<double>("caloJetPtCut", 20.0);
+    desc.add<double>("caloJetEtaCut", 3.0);
+    desc.add<bool>("doMet", true);
+    descriptions.add("scoutingCaloJetsProducer", desc);
```
Comment (Martin-Grunewald) [inline HLTrigger/JetMET/plugins/HLTScoutingCaloProducer.cc]: Please change to
```
descriptions.add("hltScoutingCaloProducer", desc);
```

File: HLTrigger/JetMET/plugins/HLTScoutingPFProducer.cc [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    desc.add<edm::InputTag>("vertexCollection", edm::InputTag("hltPixelVertices"));
+    desc.add<edm::InputTag>("metCollection", edm::InputTag("hltPFMETProducer"));
+    desc.add<edm::InputTag>("rho", edm::InputTag("hltFixedGridRhoFastjetAll"));
+    desc.add<double>("pfJetPtCut", 20.0);
+    desc.add<double>("pfJetEtaCut", 3.0);
+    desc.add<double>("pfCandidatePtCut", 0.6);
+    desc.add<bool>("doJetTags", true);
+    desc.add<bool>("doCandidates", true);
+    desc.add<bool>("doMet", true);
+    descriptions.add("scoutingPFJetsProducer", desc);
```
Comment (Martin-Grunewald) [inline HLTrigger/JetMET/plugins/HLTScoutingPFProducer.cc]: Please change to:
```
descriptions.add("hltScoutingPFProducer", desc);
```

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Pull request merged**